addons/mail/static/src/owl/components/composer_text_input/composer_text_input.js

```diff
@@ -673,26 +149,42 @@ class ComposerTextInput extends Component {
      * @param {KeyboardEvent} ev
      */
     _onKeydownEditableEscape(ev) {
-        if (this._editable.innerHTML.length !== 0) {
+        if (!this.isEmpty()) {
             return;
         }
         this.trigger('o-discard');
         ev.preventDefault();
     }
 
     /**
+     * Update the textarea selection range based on store values
+     *
+     * @private
+     */
+    _updateSelectionRange() {
+        this._textareaRef.el.setSelectionRange(
+            this.storeProps.composer.textInputCursorStart,
+            this.storeProps.composer.textInputCursorEnd,
+        );
+    }
+
+    /**
+     * Update the textarea height
      * @private
-     * @param {Event} ev
      */
-    _onSelectionChangeEditable(ev) {
-        this._saveRange();
+    _updateHeight() {
+        this._textareaRef.el.style.height = "0px";
+        this._textareaRef.el.style.height = (this._textareaRef.el.scrollHeight)+"px";
```

Comment:
space before and after operators (here `+`)

Reply:
Done